medallion/backends/taxii/memory_backend.py

```diff
@@ -113,13 +115,19 @@ def _load_file(*paths):
             return json.load(infile)
 
     def load_data_from_file(self, filename):
-        with codecs.open(filename, mode="r", encoding="utf8") as infile:
-            self.data = json.load(infile)
+        if isinstance(filename, string_types):
+            with codecs.open(filename, mode="r", encoding="utf8") as infile:
+                self.data = json.load(infile)
+        else:
+            self.data = json.load(filename)
 
     def save_data_to_file(self, filename, **kwargs):
         """The kwargs are passed to ``json.dump()`` if provided."""
-        with codecs.open(filename, mode="w", encoding="utf8") as outfile:
-            json.dump(self.data, outfile, **kwargs)
+        if isinstance(filename, string_types):
+            with codecs.open(filename, mode="w", encoding="utf8") as outfile:
```

Comment:
could we place the encoding in the function itself and pass it to io.open(). In the event someone wishes to use a different one? Can we also change to io instead of codecs -- just for consistency.